internal/rbd/rbd_util.go

```diff
@@ -2004,5 +2018,11 @@ func (rv *rbdVolume) unsetAllMetadata(keys []string) error {
 		}
 	}
 
+	err := rv.RemoveMetadata(clusterNameKey)
```

Comment:
we should not removemetadata of clusterName as clusterName is the same for both snapshots and pvc.